### DISCUSSION TITLE
TypeError: Response body object should not be disturbed or locked

### DISCUSSION THREAD

**Discussion by MinhNguyen5199** [vercel/next.js#58991] [App Router]
`// src/app/api/graphql/route.ts

import { createYoga } from 'graphql-yoga'
import { schema } from '../../../../graphql/schema'

const { handleRequest } = createYoga({
  schema,
  graphqlEndpoint: '/api/graphql',
  fetchAPI: { Response },
})

export { handleRequest as GET, handleRequest as POST}`
this block of code makes TypeError: Response body object should not be disturbed or locked.

**Comment by icyJoseph** [comment]
It has been several years since I used graphql-yoga now, but this error comes up when you do something like:

```js
// everything with the same req in scope
const data = await req.json();

// later on... 
const dataAgain = await req.json(); // ❌ the body is already read, so this is not allowed
```

I wonder if the `handleRequest` function is somehow doing that

**Reply by MinhNguyen5199** [reply]
The error happened because handlerequest was called twice as GET and POST. I dont know how to fix

**Comment by MinhNguyen5199** [comment]
I thought about it but in app router in nextjs, you cant have function
handle request, ONLY GET, POST
